extra_data/keydata.py

```diff
@@ -340,13 +340,33 @@ def as_single_value(self, rtol=1e-5, atol=0.0, reduce_by='median'):
         the first value encountered. By default, 'median' is used.
 
         If within tolerances, the reduced value is returned.
+
+        For non-numerical keys like strings, the method instead always
+        checks for uniqueness and returns such a value, if present.
         """
 
         data = self.ndarray()
 
         if len(data) == 0:
             raise NoDataError(self.source, self.key)
 
+        if not np.issubdtype(self.dtype, np.number):
+            # Handle non-numeric types first.
+
+            if reduce_by is not None:
+                raise TypeError('custom reduce method not supported for '
+                                'non-numeric type')
+
+            unique_values = np.unique(data, axis=None)
+
+            if len(unique_values) > 1:
+                raise ValueError(f'str values are not unique: {unique_values}')
```

Comment:
```suggestion
                raise ValueError(f'Values are not unique: {unique_values}')
```

Reply:
The output can potentially be very large if we display all unique values. we should maybe trim the array or not display it at all?